delfin/drivers/hpe/hpe_3par/consts.py

```diff
@@ -624,3 +626,77 @@
     '0x09f0002': 'File Persona CPG grow limit warning',
     '0x0a50001': 'File Access Auditing Alerts'
 }
+NODE_PATTERN = "^\\s*Node\\s+[-]*Name[-]*\\s+[-]*State[-]*\\s+Master\\s+"
+CPU_PATTERN = "^\\s*Node\\s+CPU\\s+[-]*Manufacturer[-]*\\s+[-]*Serial[-]*" \
+              "\\s+CPUSpeed"
+DISK_PATTERN = "^\\s*Id\\s+[-]*CagePos[-]*\\s+[-]*Type[-]*\\s+RPM\\s+State\\s+"
+DISK_I_PATTERN = "^\\s*Id\\s+[-]*CagePos[-]*\\s+[-]*State[-]*\\s+" \
+                 "[-]*Node_WWN[-]*\\s+[-]*MFR[-]*\\s+[-]*Model[-]*\\s+" \
+                 "[-]*Serial[-]*\\s+[-]*FW_Rev[-]*"
+PORT_PATTERN = "^\\s*N:S:P\\s+[-]*Mode[-]*\\s+[-]*State[-]*\\s+[-]*" \
+               "Node_WWN[-]*\\s+[-]*Port_WWN/HW_Addr[-]*\\s+"
+PORT_I_PATTERN = "^\\s*N:S:P\\s+Brand\\s+Model\\s+Rev\\s+Firmware\\s+" \
+                 "Serial\\s+HWType"
+PORT_PER_PATTERN = "^\\s*N:S:P\\s+Connmode\\s+ConnType\\s+CfgRate\\s+MaxRate"
+PORT_C_PATTERN = "^\\s*N:S:P\\s+Mode\\s+Device\\s+Pos\\s+Config\\s+" \
+                 "Topology\\s+Rate"
+PORT_ISCSI_PATTERN = "^\\s*N:S:P\\s+State\\s+IPAddr\\s+Netmask/PrefixLen\\s+" \
+                     "Gateway"
+PORT_RCIP_PATTERN = "^\\s*N:S:P\\s+State\\s+[-]*HwAddr[-]*\\s+IPAddr\\s+" \
+                    "Netmask\\s+Gateway\\s+MTU\\s+Rate"
+PORT_FCOE_PATTERN = "^\\s*N:S:P\\s+State\\s+"
+PORT_FS_PATTERN = "^\\s*N:S:P\\s+State\\s+"
+FPG_PATTERN = "^\\s*FPG\\s+[-]*Mountpath[-]*\\s+[-]*Size[-]*\\s+[-]*" \
+              "Available[-]*\\s+[-]*ActiveStates"
+CPG_PATTERN = "^\\s*Id\\s+[-]*Name[-]*\\s+Warn"
+VOLUME_PATTERN = "^\\s*Id\\s+Name\\s+Prov\\s+Compr\\s+Dedup"
+FSTORE_PATTERN = "^\\s*Fstore\\s+VFS\\s+FPG\\s+State\\s+Mode"
+FSHARE_PATTERN = "^\\s*ShareName\\s+Protocol\\s+VFS\\s+FileStore\\s+" \
+                 "ShareDir\\s+State"
+VFS_PATTERN = "^\\s*VFS\\s+FPG\\s+IPAddr\\s+State"
+IPV4_PATTERN = "^(?:[0-9]{1,3}\\.){3}[0-9]{1,3}$"
+CONTROLLER_STATUS_MAP = {
+    'OK': constants.ControllerStatus.NORMAL,
+    'NORMAL': constants.ControllerStatus.NORMAL,
+    'DEGRADED': constants.ControllerStatus.OFFLINE,
+    'FAILED': constants.ControllerStatus.OFFLINE
+}
+DISK_PHYSICAL_TYPE_MAP = {
+    'FC': constants.DiskPhysicalType.FC,
+    'SSD': constants.DiskPhysicalType.SSD,
+    'NL': constants.DiskPhysicalType.UNKNOWN
+}
+DISK_STATUS_MAP = {
+    'NORMAL': constants.DiskStatus.NORMAL,
+    'DEGRADED': constants.DiskStatus.ABNORMAL,
+    'FAILED': constants.DiskStatus.ABNORMAL,
+    'NEW': constants.DiskStatus.ABNORMAL
```

Comment:
Should it not be NORMAL?

Reply:
Disk state does not have 'query' property, only contains' NORMAL, DEGRADED, FAILED, NEW '

Reply:
I meant It was query or doubt :)

Reply:
'NORMAL': constants.DiskStatus.NORMAL,Others are classified as abnormal